### PR TITLE
Make a backup of trello FOR scripts

### PR DESCRIPTION
**Pull request by dbonattoj**
This version will change folder names by tokens (id, shortlinks, etc..) and it allows to merge with a previous backup folder in an incremental way.

It does not download again already present attachments.

eg:

trello-full-backup -b -d path/to/dir/for/the/incremental/backup
---backup---

another time:
trello-full-backup -b -d path/to/dir/for/the/incremental/backup

This will update the json and md files in each folder and download only new attachments

My usage:
I like to make a daily backup of my Trello and push it on a private repository/Dropbox. I don't want to download all the files again every time.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -140,3 +141,9 @@ Options
       -a ATTACHMENT_SIZE, --attachment-size ATTACHMENT_SIZE
                             Attachment size limit in bytes. Set to -1 to disable
                             the limit
+
+Backup mode
```
Comment (jtpio) [inline README.rst]: The script already has the word "backup" in the name. Should this feature be renamed to "incremental" instead?

File: trello_full_backup/backup.py [hunk 2]
```diff
@@ -159,6 +187,15 @@ def cli():
                         nargs='?',
                         help='Destination folder')
 
+    # Backup mode. Replace folder names by tokens and
+    # don't download the already existing attachments
+    parser.add_argument('-b', '--backup',
```
Comment (jtpio) [inline trello_full_backup/backup.py]: Could we use -i, --incremental?

File: trello_full_backup/backup.py [hunk 3]
```diff
@@ -22,6 +22,23 @@
...
+def mkdir(name):
+    ''' Make a folder if it does not exist already '''
+    if not os.access(name, os.R_OK):
+        os.mkdir(name)
+
+def get_extension(filename):
+    ''' Get the extension of a file '''
+    return os.path.splitext(filename)[1]
+
+def get_name(backup_mode, real_name, backup_name, id):
```
Comment (jtpio) [inline trello_full_backup/backup.py]: Since "id" is a Python function, rename to "card_id"?

**Comment by jtpio** [comment]
Great addition, many thanks!

Here is a summary of the few annotations I left in the diff:
- Since the script already has "backup" in the name, should the feature be called "incremental" instead? (to avoid confusion)
- In that case, the command line parameter could be named "-i, --incremental"?
- Could the shortlink feature be extracted and exposed with the "-s, --shortlink" parameter? So it can also be used in a non-incremental backup.

What do you think?

**Comment by dbonattoj** [comment]
Sure :) no problem!

**Comment by dbonattoj** [comment]
up?

**Comment by jtpio** [comment]
Looks good, gonna check it out and test it in the next days!

**Pull request merged**